applications/GeoMechanicsApplication/custom_elements/U_Pw_base_element.cpp

```diff
@@ -700,7 +702,15 @@ double UPwBaseElement<TDim, TNumNodes>::CalculateIntegrationCoefficient(
     const GeometryType::IntegrationPointsArrayType& IntegrationPoints, unsigned int PointNumber, double detJ)
 
 {
-    return IntegrationPoints[PointNumber].Weight() * detJ;
+    std::unique_ptr<StressStatePolicy> stress_state_policy;
+    if constexpr (TDim == 2) {
+        stress_state_policy = std::make_unique<PlaneStrainStressState>();
+    } else {
+        stress_state_policy = std::make_unique<ThreeDimensionalStressState>();
+    }
+
+    return stress_state_policy->CalculateIntegrationCoefficient(IntegrationPoints[PointNumber],
+                                                                detJ, GetGeometry());
```

Comment:
I know that the calculation of the integration coefficient is the same for plane strain and 3D, so this might seem overkill. However, this makes it consistent with all the other stress-state dependent functions, such that making the stress state policy a member of the elements becomes simpler.